suorituspalvelu-service/src/main/scala/fi/oph/suorituspalvelu/ui/EntityToUIConverter.scala

```diff
@@ -14,68 +15,91 @@ import scala.jdk.OptionConverters.*
 
 object EntityToUIConverter {
 
-  def convertTila(koodi: Koodi): Tila =
+  def convertVirtaTila(koodi: Koodi): Tila =
+    koodi.arvo match
+      case "1"     => Tila.KESKEN       // aktiivinen
+      case "2"     => Tila.KESKEN       // optio
+      case "3"     => Tila.VALMIS       // valmistunut
+      case "4"     => Tila.KESKEYTYNYT  // passivoitu
+      case "5"     => Tila.KESKEYTYNYT  // luopunut
+      case "6"     => Tila.KESKEYTYNYT  // päättynyt
+
```

Comment:
Enemmän ehkä food for thought kuin varsinainen välitön korjausehdotus, mutta kannattaisikohan tilan olla jotain tyyliin case class VirtaTila(tila: Tila, lahdeTila: LahdeTila), jolloin kälille voisi välittää suppiloidun arvon lisäksi myös alkuperäisen arvon jos se ehkä halutaan näyttää.